src/OrleansServiceFabricUtils/FabricMembershipOracle.cs

```diff
@@ -366,25 +373,30 @@ private void ClearCaches()
             this.multiClusterSilosCache = null;
         }
 
-        private void NotifySubscribers(SiloAddress address, SiloStatus newStatus, List<ISiloStatusListener> listeners)
+        private async Task ProcessNotifications()
         {
-            Task.Factory.StartNew(() =>
+            while (!this.notifications.IsAddingCompleted)
             {
-                foreach (var subscriber in listeners)
+                await this.notificationEvent.WaitAsync(TimeSpan.FromMinutes(1));
```

Comment:
What's the purpose of setting the 1 minute timeout? Why not just infinite? Or are could there be work to do even if no one sets the event?